src/librados/librados_c.cc

```diff
@@ -4048,7 +4048,7 @@ extern "C" int LIBRADOS_C_API_DEFAULT_F(rados_aio_write_op_operate)(
   ::ObjectOperation *oo = (::ObjectOperation *) write_op;
   librados::IoCtxImpl *ctx = (librados::IoCtxImpl *)io;
   librados::AioCompletionImpl *c = (librados::AioCompletionImpl*)completion;
-  int retval = ctx->aio_operate(obj, oo, c, ctx->snapc, translate_flags(flags));
+  int retval = ctx->aio_operate(obj, oo, c, ctx->snapc, nullptr, translate_flags(flags));
```

Comment:
This function takes `mtime` parameter, shouldn't it be propagated here?

Reply:
thanks, i didn't see that. unfortunately there's a type mismatch that prevents us from passing the `time_t*` directly as `ceph::real_time*`

we would need somewhere to store a `ceph::real_time` variable, along with a callback to convert that back to the caller's `time_t` at the end. this is much easier for the synchronous functions `rados_write_op_operate()` and `rados_write_op_operate2()`

Reply:
Based on the comment in include/rados/librados.h, yes, it should be propagated here:

* @param mtime the time to set the mtime to, NULL for the current time

However, that is a change in behavior, if only to the documented one.

Reply:
> along with a callback to convert that back to the caller's `time_t` at the end

sorry no, we only need to convert from `time_t` to `ceph::real_time`, not back

i see that [librados_cxx.cc](https://github.com/ceph/ceph/blob/0fb0dc4/src/librados/librados_cxx.cc#L93-L99) wraps the `ObjectOperation` in an `ObjectOperationImpl` that includes storage for the `real_time rt` and `real_time *prt`. it looks like librados_c.cc could do the same in `rados_create_write_op()`

Reply:
i've added 3 commits to address this. i'll work on some test coverage for it